service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -385,4 +415,44 @@ private Update createUpdateWithUpdatedAt() {
 
         return update;
     }
+
+    private boolean checkForAckOutOfOrderAndDuplicate(String conversationId) {
+        var ehrExtractStatusOptional = ehrExtractStatusRepository.findByConversationId(conversationId);
+        if (ehrExtractStatusOptional.isPresent()) {
+            EhrExtractStatus ehrExtractStatus = ehrExtractStatusOptional.get();
```

Comment:
Optional, fetching status could be refactored with orElseThrow

``` 
EhrExtractStatus ehrExtractStatus = ehrExtractStatusRepository.findByConversationId(conversationId)
            .orElseThrow(() -> new EhrExtractNonExistingException("ACK", conversationId));
            
```

the same in checkForContinueOutOfOrderAndDuplicate.